data_steward/validation/main.py

```diff
@@ -738,49 +743,58 @@ def list_submitted_bucket_items(folder_bucketitems):
     folder_bucketitems_basenames = [
         basename(file_name) for file_name in folder_bucketitems
     ]
 
-    if not _has_all_required_files(folder_bucketitems_basenames):
+    to_process_items = [
+        item for item in folder_bucketitems
+        if basename(item) not in resources.IGNORE_LIST
+    ]
+
+    if not to_process_items:
+        return files_list
+
+    # Process if all required files present
+    if _has_all_required_files(folder_bucketitems_basenames):
+        logging.info(f"All required files found, processing.")
+        return to_process_items
+
+    # Check submission times and validate if > 3 hrs old
+    logging.info(
+        f"Delaying processing for hpo_id by 3 hrs (to next cron run) "
+        f"since all required files not present. "
+        f"Missing {set(AOU_REQUIRED_FILES) - set(folder_bucketitems_basenames)}"
+    )
+
+    upper_age_threshold = min(item['timeCreated'] +
+                              datetime.timedelta(days=object_retention_days) -
+                              datetime.timedelta(days=1)
+                              for item in folder_bucketitems
+                              if basename(item) not in resources.IGNORE_LIST)
+    upper_age_threshold = upper_age_threshold.replace(tzinfo=None)
+
+    lower_age_threshold = min(item['updated'] + datetime.timedelta(hours=3)
+                              for item in folder_bucketitems
+                              if basename(item) not in resources.IGNORE_LIST)
+    lower_age_threshold = lower_age_threshold.replace(tzinfo=None)
+
+    if upper_age_threshold > utc_today:
+        if lower_age_threshold < utc_today:
+            logging.info(
+                f"All required files not found but submission is stale (> 3 hrs), processing."
+            )
+            return to_process_items
+        diff = lower_age_threshold - utc_today
+        hrs = (diff.total_seconds() // 3600) + 1
         logging.info(
             f"Delaying processing for hpo_id by 3 hrs (to next cron run) "
-            f"since all required files not present. "
-            f"Missing {set(AOU_REQUIRED_FILES) - set(folder_bucketitems_basenames)}"
-        )
-        return []
-
-    # Validate submission times
-    for item in folder_bucketitems:
-        if basename(item) not in resources.IGNORE_LIST:
-            # in common.CDM_FILES or is_pii(basename(file_name)):
-            created_date = item['timeCreated']
-            retention_time = datetime.timedelta(days=object_retention_days)
-            retention_start_time = datetime.timedelta(days=1)
-            upper_age_threshold = created_date + retention_time - retention_start_time
-            upper_age_threshold = upper_age_threshold.replace(tzinfo=None)
-
-            if upper_age_threshold > utc_today:
-                files_list.append(item)
-
-            if basename(item) in AOU_REQUIRED_FILES:
-                # restrict processing time for 5 minutes after all required files
```

Comment:
Do you know why the old logic used to wait for extra 5 minutes even if the files were all present?